### ISSUE TITLE
111

### ISSUE THREAD

**Issue by lhjlhj11** [LeiBAI/AGCRN#10]
what does the cheb_k mean?Thank you!

**Comment by HeMOua** [comment]
> what does the cheb_k mean?Thank you!

应该是Chebyshev多项式的阶数

**Comment by wy123go** [comment]
> > what does the cheb_k mean?Thank you!
> 
> 应该是Chebyshev多项式的阶数

同学，请问一下，你有用03数据集复现该模型吗

**Comment by HeMOua** [comment]
> > > what does the cheb_k mean?Thank you!
> > 
> > 
> > 应该是Chebyshev多项式的阶数
> 
> 同学，请问一下，你有用03数据集复现该模型吗

没，只用了04 07 08

**Comment by wy123go** [comment]
> > > > what does the cheb_k mean?Thank you!
> > > 
> > > 
> > > 应该是Chebyshev多项式的阶数
> > 
> > 
> > 同学，请问一下，你有用03数据集复现该模型吗
> 
> 没，只用了04 07 08
好的，感谢你的回复，我用03的，MAPE达到了几十万，也不知道是什么原因

**Comment by BellChai** [comment]
> > > > > what does the cheb_k mean?Thank you!
> > > > 
> > > > 
> > > > 应该是Chebyshev多项式的阶数
> > > 
> > > 
> > > 同学，请问一下，你有用03数据集复现该模型吗
> > 
> > 
> > 没，只用了04 07 08
> > 好的，感谢你的回复，我用03的，MAPE达到了几十万，也不知道是什么原因

应该只是数据集中出现了除0问题。修改一下MAPE的计算就好了：
把metrics.py中的MAPE_torch函数的
`return torch.mean(torch.abs(torch.div((true - pred), true)))`
改为
`mask = true>0.1`
`return ((pred[mask]-true[mask])/true[mask]).abs().sum()/(mask.sum())`
就好了

**Comment by lidexf** [comment]
> > > > > > what does the cheb_k mean?Thank you!
> > > > > 
> > > > > 
> > > > > 应该是Chebyshev多项式的阶数
> > > > 
> > > > 
> > > > 同学，请问一下，你有用03数据集复现该模型吗
> > > 
> > > 
> > > 没，只用了04 07 08
> > > 好的，感谢你的回复，我用03的，MAPE达到了几十万，也不知道是什么原因
> 
> 应该只是数据集中出现了除0问题。修改一下MAPE的计算就好了： 把metrics.py中的MAPE_torch函数的 `return torch.mean(torch.abs(torch.div((true - pred), true)))` 改为 `mask = true>0.1` `return ((pred[mask]-true[mask])/true[mask]).abs().sum()/(mask.sum())` 就好了

那忽略掉真实值mask = true>0.1的情况这样合理吗？

**Comment by lidexf** [comment]
> > > > > > what does the cheb_k mean?Thank you!
> > > > > 
> > > > > 
> > > > > 应该是Chebyshev多项式的阶数
> > > > 
> > > > 
> > > > 同学，请问一下，你有用03数据集复现该模型吗
> > > 
> > > 
> > > 没，只用了04 07 08
> > > 好的，感谢你的回复，我用03的，MAPE达到了几十万，也不知道是什么原因
> 
> 应该只是数据集中出现了除0问题。修改一下MAPE的计算就好了： 把metrics.py中的MAPE_torch函数的 `return torch.mean(torch.abs(torch.div((true - pred), true)))` 改为 `mask = true>0.1` `return ((pred[mask]-true[mask])/true[mask]).abs().sum()/(mask.sum())` 就好了

忽略掉真实值小于等于0.1，这样合理吗？

**Comment by lidexf** [comment]
论文中，使用的切比雪夫多项式K=2，可是
        for k in range(2, self.cheb_k):
            support_set.append(torch.matmul(2 * supports, support_set[-1]) - support_set[-2])这里循环时要求至少等于3
